### DISCUSSION TITLE
Malicious/virus uploads to MinIO

### DISCUSSION THREAD

**Discussion by dthemg** [minio/minio#15651] [Q&A] [answered]
Hi

I would like users to be able to upload files to my MinIO server. 

Of course, I want to make sure that no malicious uploads are made. Before uploading to MinIO, I do the recommended stuff:
* Rename the file to something generated
* Run ClamAV on the file
* Check the file extension
* Check the file size

However, I'd like to know what happens if a malicious piece of code still manages to bypass all this. Could such a file cause any damage on the MinIO server? Are the uploaded objects stored in an environment with execution rights?

Best regards

David

**Comment by klauspost** [comment, marked as answer]
>  Could such a file cause any damage on the MinIO server? 

No. No object data is ever executed. That would be considered a critical security issue.

> Are the uploaded objects stored in an environment with execution rights?

No. Files are written with 0666 (r+w) permissions. In both distributed and Single-Node Single-Drive installs content is modified before it is written. Furthermore you can add [server-side encryption](https://docs.min.io/minio/baremetal/security/encryption-overview.html) to encrypt the data before it is written.

**Reply by dthemg** [reply]
Thanks for the answer @klauspost!
